### PR TITLE
feat: add Datadog crashreporting

### PR DESCRIPTION
**Pull request by netbe**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

There are no crash reporter within the app.

### Solutions

Add crash reporting to Datadog

#### How to Test

- Tap on crash app button within developer tools

### Notes (Optional)

Need to add fastlane_plugin or script to upload dsym files to datadog

### Attachments (Optional)

Screenshot 2023-01-20 at 10 18 30

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: Wire-iOS.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1828,10 +1830,10 @@
 			dstPath = "";
 			dstSubfolderSpec = 10;
 			files = (
+				013718262979AC2600BD961F /* DatadogCrashReporting.xcframework in Embed Frameworks */,
 				0139E49C296EE1E200E17ADE /* Datadog.xcframework in Embed Frameworks */,
 				A96867AD26A5E59F00679D95 /* WireDataModel.xcframework in Embed Frameworks */,
 				A96867CE26A5E68600679D95 /* Down.xcframework in Embed Frameworks */,
-				01AAD75D294FD4CF00F1CFFD /* Datadog.xcframework in Embed Frameworks */,
```
Comment (agisilaos) [inline Wire-iOS.xcodeproj/project.pbxproj:1834]: Is that intended?

**Review by agisilaos — APPROVED** [review]

File: Wire-iOS.xcodeproj/project.pbxproj (hunk 1 above)
Comment (netbe) [reply Wire-iOS.xcodeproj/project.pbxproj:1834]: I am checking

**Review by tmspzz — APPROVED** [review]

File: Wire-iOS.xcodeproj/project.pbxproj (hunk 1 above)
Comment (netbe) [reply Wire-iOS.xcodeproj/project.pbxproj:1834]: seems to be ok

File: Wire-iOS/Sources/Managers/DatadogWrapper.swift [hunk 2]
```diff
@@ -69,6 +70,7 @@ public class DatadogWrapper {
                 .trackUIKitRUMActions()
                 .trackBackgroundEvents()
                 .trackRUMLongTasks()
+                .enableCrashReporting(using: DDCrashReportingPlugin()) // NEW
```
Comment (johnxnguyen) [inline Wire-iOS/Sources/Managers/DatadogWrapper.swift:73]:
```suggestion
                .enableCrashReporting(using: DDCrashReportingPlugin())
```

**Review by johnxnguyen — APPROVED** [review]

**Pull request merged**